### ISSUE TITLE
[Electron][A weird error that happens when the main process started for a while]

### ISSUE THREAD

**Issue by Bistard** [Bistard/nota#126]
ERROR LOG：
[4124:0806/034502.690:ERROR:gpu_init.cc(454)] Passthrough is not supported, GL is disabled, ANGLE is

**Comment by snnsnn** [comment]
If I remember correctly this is GPU related error, there should be an electron flag like --disable-gpu to suppress this.

I've come accross your project when I was looking for the other nota project: https://github.com/nota-lang/nota

I've also implemented VSCode like app and closely familiar with the VSCode code base. Here are some more projects that are inspired by VSCode which may help you for ideas:

- https://github.com/opensumi
- https://github.com/eclipse-theia/theia
- https://github.com/joelday/decoration-ioc

**Comment by Bistard** [comment]
@snnsnn Thx for leaving a comment, due to the fact the app continued to work as expected after the error appeared, I haven't worried about fixing this one. I believe it is somehow related to hardware acceleration in Chromium.

Also, appreciate for sharing those great repositories with us :-D We already had a very strong and stable software infrastructure in the last two years. But it would be cool to look up for those repo for good design as well.

Thanks again for commenting👍 We are still developing it and haven't reach an alpha version yet.

**Comment by Bistard** [comment]
@snnsnn Speaking of the [ioc solution](https://github.com/joelday/decoration-ioc) you posted, I re-implement the same DI system from the VSCode just like the repo u posted. It is an interesting system to implement :)

**Comment by snnsnn** [comment]
@Bistard Yes, I posted the link because it took me a while find out how everything was glued together. :) Glad you liked it. There is no documentation on how VSCode is architected other than a few wiki pages: https://github.com/microsoft/vscode/wiki

VSCode implementation was very complex for my use case too,  I also re-wrote it, but in a slightly different way, after checking many alternatives including [inversify](https://github.com/inversify/InversifyJS), [tsyringe](https://github.com/microsoft/tsyringe), [typed-inject](https://github.com/nicojs/typed-inject) and many others. I used [awilix](https://github.com/jeffijoe/awilix) initially, which I liked a lot, then re-implemented myself with a more streamlined execution path and a better typescript support, and achieved about 2x speed boost.

If you are into VSCode like app architecture, you can take a look at Theia's documentation, it will help you a lot: https://theia-ide.org/docs/
